### ISSUE TITLE
[dynamo] Proxy conversion failure in DETR/detection models (54 models, 234 breaks)

### ISSUE THREAD

**Issue by penguinwu** [penguinwu/oss-model-graph-break-corpus#8]
## Summary

54 models produce 234 graph breaks from this pattern.

## Break Reason

```
Graph Break Reason: Encountered graph break when attempting to trace CALL: a function call, e.g. f(x, y):
```

## Affected Models

| Model | Modes | Breaks |
|-------|-------|--------|
| AriaForConditionalGeneration | eval, train | 2 |
| AriaModel | eval, train | 2 |
| AyaVisionForConditionalGeneration | eval, train | 2 |
| Cohere2VisionForConditionalGeneration | eval, train | 2 |
| DFineModel | eval, train | 14 |
| DeepseekVLForConditionalGeneration | eval, train | 2 |
| Ernie4_5_VLMoeForConditionalGeneration | eval, train | 2 |
| Ernie4_5_VL_MoeForConditionalGeneration | eval, train | 2 |
| Florence2ForConditionalGeneration | eval, train | 2 |
| Florence2Model | eval, train | 2 |
| Gemma3ForConditionalGeneration | eval, train | 2 |
| Gemma4-vision | eval | 1 |
| Gemma4ForConditionalGeneration | eval, train | 2 |
| Glm46VForConditionalGeneration | eval, train | 2 |
| Glm4vForConditionalGeneration | eval, train | 2 |
| Glm4vMoeForConditionalGeneration | eval, train | 2 |
| GlmOcrForConditionalGeneration | eval, train | 2 |
| GotOcr2ForConditionalGeneration | eval, train | 2 |
| GotOcr2Model | eval, train | 2 |
| GroundingDinoModel | eval, train | 20 |
| InternVLForConditionalGeneration | eval, train | 2 |
| InternVLModel | eval, train | 2 |
| JanusForConditionalGeneration | eval, train | 2 |
| Lfm2VlForConditionalGeneration | eval, train | 2 |
| LightOnOcrForConditionalGeneration | eval, train | 2 |
| Llama4ForConditionalGeneration | eval, train | 2 |
| LlavaForConditionalGeneration | eval, train | 2 |
| LlavaNextForConditionalGeneration | eval, train | 2 |
| LlavaNextVideoForConditionalGeneration | eval, train | 2 |
| LlavaOnevisionForConditionalGeneration | eval, train | 2 |
| LwDetrModel | eval, train | 20 |
| MMGroundingDinoModel | eval, train | 20 |
| Mistral3ForConditionalGeneration | eval, train | 2 |
| Ovis2ForConditionalGeneration | eval, train | 2 |
| PPDocLayoutV2Model | eval, train | 12 |
| PPDocLayoutV3Model | eval, train | 12 |
| PaddleOCRVLForConditionalGeneration | eval, train | 2 |
| PaliGemmaForConditionalGeneration | eval, train | 2 |
| PaliGemmaModel | eval, train | 2 |
| Qwen2AudioForConditionalGeneration | eval, train | 2 |
| Qwen2VLForConditionalGeneration | eval, train | 2 |
| Qwen2_5OmniThinkerForConditionalGeneration | eval, train | 2 |
| Qwen2_5_VLForConditionalGeneration | eval, train | 2 |
| Qwen3OmniMoeThinkerForConditionalGeneration | eval, train | 2 |
| Qwen3VLForConditionalGeneration | eval, train | 2 |
| Qwen3VLMoeForConditionalGeneration | eval, train | 2 |
| Qwen3_5ForConditionalGeneration | eval, train | 2 |
| Qwen3_5MoeForConditionalGeneration | eval, train | 2 |
| RTDetrModel | eval, train | 12 |
| RTDetrV2Model | eval, train | 14 |
| SmolVLMForConditionalGeneration | eval, train | 6 |
| VideoLlama3ForConditionalGeneration | eval, train | 2 |
| VipLlavaForConditionalGeneration | eval, train | 2 |
| VitsModel | eval, train | 17 |

## Impact

Fixing this pattern moves **2 models** to fullgraph (models where this is the only break reason).

## Related Issues

- #2
- #10
- #11
- #17
- #18
- #23
- #24
- #54
- #55
- #56

## Tested On

- PyTorch 2.13.0.dev20260425+cu126
- Transformers 5.6.2
- Sweep date: 2026-04-26T08:48:25

**Comment by penguinwu** [comment]
## Nightly Validation (PyTorch 2.12.0.dev20260407)

**Status: Still present in nightly.** Proxy conversion failures in DETR-family models persist.

- Tested against: PyTorch 2.12.0.dev20260407+cu128, Transformers 5.5.0
- No change from 2.10 → 2.11 → nightly — 193 breaks across 9 models

**Cross-version trend:** Stable across all tested versions (2.8–nightly).

**Comment by penguinwu** [comment]
## Sweep update — 2026-04-28

Verified on: `torch 2.13.0a0+gitf8d66d2` (source build, includes PR #181552) · transformers `5.6.2` · `fullgraph=True, dynamic=True`

**Status: Partial progress — 1/53 models pass fullgraph, 28 still breaking**

**Now passing fullgraph:**
- ✅ `SmolVLMForConditionalGeneration`

**Changed status (infra error, not graph-break):** `DFineModel`, `Ernie4_5_VLMoeForConditionalGeneration`, `Ernie4_5_VL_MoeForConditionalGeneration`, `Glm46VForConditionalGeneration`, `Glm4vForConditionalGeneration`, `Glm4vMoeForConditionalGeneration`, `GlmOcrForConditionalGeneration`, `GroundingDinoModel`, `LwDetrModel`, `MMGroundingDinoModel`, `Mistral3ForConditionalGeneration`, `PPDocLayoutV2Model`, `PPDocLayoutV3Model`, `PaddleOCRVLForConditionalGeneration`, `Qwen2VLForConditionalGeneration`, `Qwen2_5_VLForConditionalGeneration`, `Qwen3VLForConditionalGeneration`, `Qwen3VLMoeForConditionalGeneration`, `Qwen3_5ForConditionalGeneration`, `Qwen3_5MoeForConditionalGeneration`, `RTDetrModel`, `RTDetrV2Model`, `VideoLlama3ForConditionalGeneration`, `VitsModel`

**Still breaking:**
- ❌ `AriaForConditionalGeneration`
- ❌ `AriaModel`
- ❌ `AyaVisionForConditionalGeneration`
- ❌ `Cohere2VisionForConditionalGeneration`
- ❌ `DeepseekVLForConditionalGeneration`
- ❌ `Florence2ForConditionalGeneration`
- ❌ `Florence2Model`
- ❌ `Gemma3ForConditionalGeneration`
- ... and 20 more

**Minimal repro** (`AriaForConditionalGeneration`):
```bash
# In ~/projects/oss-model-graph-break-corpus, with venv activated:
echo '["AriaForConditionalGeneration"]' > /tmp/repro.json
python3 sweep/run_sweep.py sweep \
  --models /tmp/repro.json \
  --modes eval \
  --compile-kwargs '{"fullgraph": true, "dynamic": true, "backend": "eager"}' \
  --setup-script sweep/configs/animesh-logging-suppress.py
```

Break reason: `Encountered graph break when attempting to trace CALL_FUNCTION_EX: a variadic function call, e.g. f(*args):`

**Comment by penguinwu** [comment]
**Status update — corrected classification (2026-04-28 evening)**

Earlier today's per-issue walk on the [Animesh fullgraph sweep](https://github.com/penguinwu/oss-model-graph-break-corpus/issues/70) classified several models in this issue's affected list as `eager_error`. On further investigation we found that classification was masking the real status — the source-build venv (`torch-181552`) was missing `libnvrtc-builtins.so.12.4`, causing first-eager-call CUDA reduce kernel JIT to fail. That blocked the fullgraph compile from running; the sweep recorded `eager_error` instead of the real `full_graph` / `graph_break` verdict.

After fixing the venv (commit [`50a3d3d`](https://github.com/penguinwu/oss-model-graph-break-corpus/commit/50a3d3d) + [`e82f93b`](https://github.com/penguinwu/oss-model-graph-break-corpus/commit/e82f93b) — `nvidia-cuda-*-cu12` install + sitecustomize.py preload + USE_LAPACK + timm install) and re-running just the affected models, the corrected status is:

**NEW fullgraph wins (9 work items)** — these were `eager_error` in the morning sweep but actually pass fullgraph under torch with PR #181552:

| Model | Mode |
|---|---|
| DFineModel | eval |
| GroundingDinoModel | eval |
| LwDetrModel | eval |
| LwDetrModel | train |
| MMGroundingDinoModel | eval |
| PPDocLayoutV2Model | eval |
| PPDocLayoutV3Model | eval |
| RTDetrModel | eval |
| RTDetrV2Model | eval |

**Confirmed graph_break under fullgraph (27 work items)** — these were `eager_error` in the morning sweep, now correctly classified as still-breaking (relevant input for this issue):

`DFineModel/train`, `Glm46V/Glm4v/Glm4vMoe/GlmOcrForConditionalGeneration` (all 4 × eval+train), `GroundingDinoModel/train`, `MMGroundingDinoModel/train`, `Mistral3ForConditionalGeneration` (eval+train), `PPDocLayoutV2Model/train`, `PPDocLayoutV3Model/train`, `PaddleOCRVLForConditionalGeneration` (eval+train), `Qwen2VLForConditionalGeneration` (eval+train), `Qwen3VLForConditionalGeneration` (eval+train), `Qwen3VLMoeForConditionalGeneration` (eval+train), `RTDetrModel/train`, `RTDetrV2Model/train`, `VideoLlama3ForConditionalGeneration` (eval+train).

**Note on train-mode for DETR-class:** train-mode usually exercises additional code paths (loss, target matching) that surface breaks eval skips. LwDetrModel is the only DETR-class model with both modes passing fullgraph here.

**Repro for any of the above:**
```bash
# eval (replace MODEL with one from the affected list)
python tools/reproduce.py --mode eval --device cuda <MODEL>
# fullgraph + dynamic config used by the sweep:
python sweep/run_sweep.py sweep --models '[{"name":"<MODEL>","source":"hf"}]' \
    --modes eval train --workers 1 --timeout 180 \
    --output-dir /tmp/repro_<MODEL> \
    --compile-kwargs '{"fullgraph": true, "dynamic": true, "backend": "eager"}'
```

Filed by Otter on Peng's behalf as part of corpus-side hygiene cleanup. A separate timm-cluster re-run is in flight (~14 more models that may also flip).

**Comment by penguinwu** [comment]
**Follow-up — additional context from timm-cluster re-run (2026-04-28 evening, after main update above)**

After the venv hygiene fixes ([commits `50a3d3d` + `e82f93b`](https://github.com/penguinwu/oss-model-graph-break-corpus/commit/e82f93b)), we re-tested another 14 models that were `create_error` in the morning sweep due to missing `timm`. *None of these models appear in this issue's affected-models list* (they were create-errored before being testable), but several are DETR-family and worth surfacing here for context:

**DETR-family models that pass fullgraph (eval) under torch with PR #181552 — newly testable after `pip install timm`:**

| Model | eval | train |
|---|---|---|
| `DetrModel` | full_graph | timeout (likely large model) |
| `ConditionalDetrModel` | full_graph | graph_break |
| `DabDetrModel` | full_graph | full_graph |
| `DeformableDetrModel` | full_graph | graph_break |
| `TableTransformerModel` | full_graph | graph_break |

Other timm-cluster wins (not DETR-class): `FastVlmModel`, `FastVlmForConditionalGeneration`, `PeVideoModel`, `PerceptionLMModel`, `PerceptionLMForConditionalGeneration`, `TimmWrapperModel` (all eval+train where indicated).

These were not regressions — they were always create-errored, so the issue tracker simply never saw them. Now part of the testable baseline. The corpus repo's `sweep_results/experiments/animesh-fullgraph-2026-04-28/` directory has the morning's data; the re-run results live in `animesh-relaunch-19models-2026-04-28/` and `animesh-relaunch-timm-2026-04-28/`.

Filed by Otter on Peng's behalf.

**Comment by penguinwu** [comment]
## Closing this issue — superseded by per-root-cause sub-issues

**Reason:** This umbrella was too coarse to be actionable. The "trace CALL" / "Encountered graph break when attempting to trace CALL" pattern is just the bytecode-instruction wrapper — the underlying root causes are 8+ distinct things, each with its own resolution path and its own owner.

Per the umbrella-split policy (decided 2026-05-04): large umbrella issues bundle multiple root causes and never close, hiding actual progress. Splitting into specific issues per-root-cause makes progress measurable (= number of closed issues).

## What this umbrella actually contained (550 breaks across 29 still-broken models, sweep 2026-05-03)

| Count | Root cause | Filed as |
|---|---|---|
| 246× | `Tensor.item()` with `capture_scalar_outputs=False` | #115 |
| 182× | `transformers/utils/output_capturing.py` (`lock` ctx + `ContextVar.get`) | #113 |
| 88× | `setattr` builtin not traced (concentrated at `torch/utils/hooks.py:27`) | #114 |
| 85× | `torch._check_with` msg_callable closure constraint at `import_utils.py:1540` | #112 |
| 12× | `range(int, Tensor, int)` builtin | #116 |
| 168× | `Operator aten._local_scalar_dense.default` | #55 (already exists) |
| 130× | `Operator aten.nonzero.default` | #18 (already exists) |
| 14× | Data-dependent branching | #18 (already exists) |
| 12× | `Operator aten.repeat_interleave.Tensor` | #18 (already exists) |

(Numbers above are corpus-wide for the new sub-issues, but only #8's named-model subset for the entries that move to existing issues.)

## What progress this revealed

5 of the original 34 named models in this issue ARE now `full_graph` in both modes (verified 2026-05-03 sweep): LwDetrModel, MllamaVisionModel, Siglip2Model, Siglip2VisionModel, SmolVLMForConditionalGeneration. **Attribution: torch (Dynamo) — verified by re-running with current harness + older torch + baseline transformers, which reproduces the original break.**

That progress will be tracked going forward in the new per-root-cause sub-issues, where each closure is a real win.

Closing as superseded.
